crates/bevy_math/src/curves/fixed.rs

```diff
@@ -0,0 +1,121 @@
+use crate::{
+    curves::{Curve, CurveCursor},
+    interpolation::Lerp,
+};
+
+// TODO: impl Serialize, Deserialize
+/// Curve with evenly spaced keyframes, in another words a curve with a fixed frame rate.
+///
+/// This curve maintains the faster sampling rate over a wide range of frame rates, because
+/// it doesn't rely on keyframe cursor. As a downside, it will have a bigger memory foot print.
+#[derive(Default, Debug)]
+pub struct CurveFixed<T> {
+    frame_rate: f32,
+    /// Negative number of frames before the curve starts
+    negative_offset: f32,
+    pub keyframes: Vec<T>,
+}
+
+impl<T: Clone> Clone for CurveFixed<T> {
+    fn clone(&self) -> Self {
+        Self {
+            frame_rate: self.frame_rate,
+            negative_offset: self.negative_offset,
+            keyframes: self.keyframes.clone(),
+        }
+    }
+}
+
+impl<T> CurveFixed<T> {
+    pub fn from_keyframes(frame_rate: f32, offset: isize, keyframes: Vec<T>) -> Self {
+        Self {
+            frame_rate,
+            negative_offset: -(offset as f32),
```

Comment:
What's the purpose of having `offset` as an `isize` and not a `f32`?
The conversion to and from `f32` and not the best imo.

Reply:
I had a reason but in a second though maybe I should just use a `f32`

Reply:
I remember negative_offset should be called negative_frame_offset it's an integer stored in the `f32` for performance reasons

Reply:
there's a similar issue above, closing this one